setuptools/_distutils/dist.py

```diff
@@ -23,6 +23,7 @@
     DistutilsArgError,
     DistutilsClassError,
 )
+from setuptools.extern.packaging.utils import canonicalize_name, canonicalize_version
```

Comment:
I see. Am I understanding correctly that https://github.com/pypa/distutils has no dependencies, so we can't use `packaging` there at all?

> Maybe it would be better to go down in the class hierarchy to avoid modifying this directory directly?

Annoying, but makes sense I guess!

Reply:
> we can't use `packaging` there at all?

I believe this assessment is correct 😅.